pkg/prometheus/rules.go

```diff
@@ -208,34 +216,8 @@ func (c *Operator) selectRules(p *monitoringv1.Prometheus, namespaces []string)
 	return rules, nil
 }
 
-func generateContent(promRule monitoringv1.PrometheusRuleSpec, enforcedNsLabel, ns string) (string, error) {
-	if enforcedNsLabel != "" {
-		for gi, group := range promRule.Groups {
-			group.PartialResponseStrategy = ""
-			for ri, r := range group.Rules {
-				if len(promRule.Groups[gi].Rules[ri].Labels) == 0 {
-					promRule.Groups[gi].Rules[ri].Labels = map[string]string{}
-				}
-				promRule.Groups[gi].Rules[ri].Labels[enforcedNsLabel] = ns
-
-				expr := r.Expr.String()
-				parsedExpr, err := parser.ParseExpr(expr)
-				if err != nil {
-					return "", errors.Wrap(err, "failed to parse promql expression")
-				}
-				err = injectproxy.SetRecursive(parsedExpr, []*labels.Matcher{{
-					Name:  enforcedNsLabel,
-					Type:  labels.MatchEqual,
-					Value: ns,
-				}})
-				if err != nil {
-					return "", errors.Wrap(err, "failed to inject labels to expression")
-				}
-
-				promRule.Groups[gi].Rules[ri].Expr = intstr.FromString(parsedExpr.String())
-			}
-		}
-	}
+func generateContent(promRule monitoringv1.PrometheusRuleSpec) (string, error) {
+
 	content, err := yaml.Marshal(promRule)
```

Comment:
I don't see us doing this previously, so I'm fine with this being a follow up, but we need to discard the `partial_response_strategy` (also shouldn't be snake case).

Reply:
you do - see [pkg/prometheus/rules.go#L214](https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/rules.go#L214)

snake case is an issue of Thanos unfortunately. See [partial-response](https://github.com/thanos-io/thanos/blob/v0.4.0-rc.0/docs/components/rule.md#partial-response)

but it is an issue - I'm discarding this field for Thanos Rules also what I shouldn't do - will fix it